syntax="proto3";

package distributed_agenda;
option go_package = "./server/proto";

enum OperationOutcome{
    SUCCESS = 0;
    FAILED = 1;
}

message User{
    string username = 1;
    string name = 2;
    string passwordHash = 3;
    string email = 4;
}

message GetUserRequest{
    string username = 1;
}

message EditUserRequest{
    User user = 1;
}

message GetUserResponse{
    User user = 1;
}

message EditUserResponse{
    OperationOutcome result = 1;
}

service UserService{

    rpc GetUser(GetUserRequest) returns (GetUserResponse){}

    rpc EditUser(EditUserRequest) returns (EditUserResponse){}
}